Source/JavaScriptCore/bytecompiler/BytecodeGenerator.cpp

```diff
@@ -4937,6 +4937,19 @@ void BytecodeGenerator::emitRequireObjectCoercible(RegisterID* value, ASCIILiter
     emitLabel(target.get());
 }
 
+void BytecodeGenerator::emitRequireObjectCoercibleForDestructuring(RegisterID* value, const Identifier* propertyName)
+{
+    Ref<Label> target = newLabel();
+    OpJnundefinedOrNull::emit(this, value, target->bind(this));
+
+    if (propertyName && !propertyName->isNull())
+        emitThrowTypeError(Identifier::fromString(m_vm, makeString("Cannot destructure property '"_s, propertyName->string(), "' from null or undefined value"_s)));
+    else
+        emitThrowTypeError("Cannot destructure null or undefined value"_s);
```

Comment:
Slightly annoying to have the message be vague: "null or undefined". Why not have separate messages for null and undefined? Would emitting an extra branch in the throwing case be too much of a waste?

Reply:
I also considered distinguishing between null and undefined checks. However, checking them separately would increase the number of bytecode instructions.